### ISSUE TITLE
Block: How to run a js runtime along with tokio task?

### ISSUE THREAD

**Issue by linrongbin16** [rsvim/rsvim#112]
In #109 , I implement a dead simple js runtime with a `line_wrap` option.

I try to initialize it in the main thread, and spawn it to run in tokio's task. But there's an error says `v8::Isolate` (which should be the memory owner of the whole js runtime) doesn't implement `std::marker::Send` so it cannot be sent between threads. Which means I cannot throw it into a tokio thread.

<img width="1539" alt="image" src="https://github.com/user-attachments/assets/1f44d780-d879-4446-a1de-e683fb7ff846">

Also see this comment in rusty_v8: https://github.com/denoland/rusty_v8/issues/1467#issuecomment-2348066083

PR [rusty_v8 - #1411](https://github.com/denoland/rusty_v8/pull/1411) provide `Locker` for make Isolate able to be sent between threads.

**Comment by linrongbin16** [comment]
With a second thought, is it a must to run/execute js scripts in a non-main thread? Can we directly run it in the main loop?

As users scripts growing bigger and bigger, simply loading these scripts and plugins can take a lot of time, that's why we want to use another thread to load them. i.e. let user first open the editor UI and move their cursor on the window, and loading and enabling the configurations/plugins at the same time. So we still need to load user scripts in another thread, not the main thread.

But in the implementation, we could try to completely give up the main thread to js runtime. Which involves a lot of code changes, hope the PR 1411 could get merged into rusty_v8, so we could use it directly.

**Comment by linrongbin16** [comment]
Now I understand why emacs directly fork a child process to run javascript, it's so difficult....

**Comment by linrongbin16** [comment]
Resolved in #116 .

**Comment by Earthmark** [comment]
If you're working with Tokyo you can use a local set to spawn a task that's not send. It has issues, like it can't split effort between threads, but it will let the js code actually run. 

I have an example of this in https://github.com/Earthmark/spooler/blob/main/spooler%2Fsrc%2Flogic%2Fworker.rs

Where I used very explicit managed worker thread pools to run multiple isolates at once on the same thread, and have multiple isolate-threads going in parallel. 

Warning: this code is very old and probably won't work, but I doubt tokio dropped local set support.

**Comment by linrongbin16** [comment]
hi @Earthmark, really thanks to you! I will take a look!

**Comment by linrongbin16** [comment]
Some discussions about V8 and tokio runtime:

- https://github.com/denoland/deno/discussions/25880
- https://github.com/denoland/deno/discussions/25799
- https://github.com/aalykiot/dune/issues/227

**Comment by linrongbin16** [comment]
Module map related:
- https://github.com/aalykiot/dune/issues/307
